### PR TITLE
add rate and review

### PR DESCRIPTION
**Pull request by hieuhtk58**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      redirect_to book_path(@book)
+    else
+      render "new"
+    end
+  end
+
+  private
+
+    def review_params
+      params.require(:review).permit(:rating, :content)
```
Comment (ngotungson) [inline app/controllers/reviews_controller.rb]: remove ()

File: config/routes.rb [hunk 2]
```diff
@@ -9,7 +9,10 @@
   post "login" => "sessions#create"
   delete "logout" => "sessions#destroy"
 
-  resources :books, only: [:index, :show]
+  resources :users
+  resources :books, only: [:index, :show] do
+    resources :reviews
```
Comment (ngotungson) [inline config/routes.rb]: giới hạn routes

**Comment by phanbt58** [comment]
ok

**Comment by ngotungson** [comment]
OK

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  def new
+    @review = @book.reviews.build
+  end
+
+  def create
+    @review = @book.reviews.build review_params
+    if @review.save
+      redirect_to book_path(@book)
+    else
+      render "new"
```
Comment (thangtx88) [inline app/controllers/reviews_controller.rb]: render :new

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  before_action :load_book
+
+  def new
+    @review = @book.reviews.build
+  end
+
+  def create
+    @review = @book.reviews.build review_params
+    if @review.save
+      redirect_to book_path(@book)
```
Comment (nhunght) [inline app/controllers/reviews_controller.rb]: remove ()

File: app/controllers/reviews_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  before_action :load_book
+
+  def new
+    @review = @book.reviews.build
+  end
+
+  def create
+    @review = @book.reviews.build review_params
+    if @review.save
+      redirect_to book_path@book
```
Comment (nhunght) [inline app/controllers/reviews_controller.rb]: redirect_to book_path @book

**Pull request merged**